refactor: update stock module onboarding

18alantom opened the pull request:
### Description
Updation of Stock module onboarding.
![Screen Shot 2021-06-18 at 14 45 39](https://user-images.githubusercontent.com/29507195/122538123-de8b7600-d043-11eb-9654-fb5e9df58cbc.png)

18alantom committed:
refactor: update stock onboarding

18alantom committed:
refactor: add form tour for stock module onboarding

18alantom committed:
refactor: move trailing whitespace out of translate func

18alantom committed:
refactor: sider/semgrep

rmehta commented:
This needs more explanation

1. Add Item Group
2. Remove Delivery Note / Purchase Receipt - they cannot be part of on boarding

rmehta commented on erpnext/stock/doctype/delivery_note/delivery_note.js:
```diff
@@ -353,3 +353,23 @@ erpnext.stock.delivery_note.set_print_hide = function(doc, cdt, cdn){
+	{
+		fieldname: "customer",
+		title: __("Customer"),
+		description: __("This field is used to set the 'Customer'.")
```
Show, not tell.

Suggestion: "Select the Customer who is being delivered to"

rmehta commented on erpnext/stock/doctype/delivery_note/delivery_note.js:
```diff
@@ -353,3 +353,23 @@ erpnext.stock.delivery_note.set_print_hide = function(doc, cdt, cdn){
+	{
+		fieldname: "items",
+		title: __("Items"),
+		description: __("This table is used to set details about the 'Item', 'Qty', 'Basic Rate', etc.") + " " +
```
"Select Items to be delivered"

rmehta commented on erpnext/stock/doctype/delivery_note/delivery_note.js:
```diff
@@ -353,3 +353,23 @@ erpnext.stock.delivery_note.set_print_hide = function(doc, cdt, cdn){
+	{
+		fieldname: "set_posting_time",
+		title: __("Edit Posting Date and Time"),
+		description: __("This option can be checked to edit the 'Posting Date' and 'Posting Time' fields.")
```
This is not important

rmehta commented on erpnext/stock/doctype/delivery_note/delivery_note.js:
```diff
@@ -353,3 +353,23 @@ erpnext.stock.delivery_note.set_print_hide = function(doc, cdt, cdn){
+		title: __("Customer"),
+		description: __("This field is used to set the 'Customer'.")
+	},
+	{
```
Next step would be to select the Sales Order for which delivery is being made.

rmehta reviewed (changes requested):
Is there a design spec for this? @umairsy

umairsy commented on erpnext/stock/doctype/item/item.js:
```diff
@@ -795,4 +795,43 @@ frappe.ui.form.on("UOM Conversion Detail", {
+	{
+		fieldname: "stock_uom",
+		title: __("Default Unit of Measure"),
+		description: __("The Item Code can be thought of as a short-form that refers to the item.") + " " +
```
This Description repeation. It could be:

"It's a Unit of Measurement in which stock balance will be maintained for this Item."

Am further updating Stock Onboarding and Description in this doc. Let's catch-up quickly to discuss further.

https://docs.google.com/document/d/1O2hGLobhVUdzZKq_P_EujfqxDi9pRx_hI0hnuYWdbAk/edit#

umairsy commented on erpnext/stock/onboarding_step/create_a_supplier/create_a_supplier.json:
```diff
@@ -1,18 +1,19 @@
 {
- "action": "Create Entry",
+ "action": "Show Form Tour",
```
Is Tour for Supplier missing? Or it's being worked upon in other module? @18alantom @rohitwaghchaure

18alantom replied:
Since Supplier is present in other two modules (Buying and Accounting), it's been removed from here to avoid repetition.

18alantom committed:
refactor: remove DN, PR; change wording, add/remove steps in tour

18alantom committed:
Merge branch 'version-13-hotfix' into update-stock-onboarding

18alantom committed:
refactor: add watch video step for stock opening balance

18alantom committed:
refactor: reorder steps according to stock settings refactor

18alantom committed:
refactor: fix typo, remove target warehouse cause SE Type dependency

18alantom commented:
**Notes/reasoning regarding changes**
- Removed Delivery Note, Purchase Receipt: Too much complexity for onboarding.
- Removed Supplier: Already present in two onboardings, ∴ to avoid repetition.
- Shifted Stock Settings to after Warehouse: This step needs context, Item and Warehouse steps provide context.
- Added Stock Opening Balance video at the end: Stock Reconciliation is too complex for onboarding, but information about Stock Opening Balance is required, hence the video.

(@umairsy  please check these 👆)

18alantom committed:
fix: semgrep, remove trailing and leading whitespaces

ankush commented on erpnext/hooks.py:
```diff
@@ -425,6 +425,7 @@
 		'erpnext.controllers.taxes_and_totals.get_regional_round_off_accounts': 'erpnext.regional.india.utils.get_regional_round_off_accounts',
 		'erpnext.hr.utils.calculate_annual_eligible_hra_exemption': 'erpnext.regional.india.utils.calculate_annual_eligible_hra_exemption',
 		'erpnext.hr.utils.calculate_hra_exemption_for_period': 'erpnext.regional.india.utils.calculate_hra_exemption_for_period',
+		'erpnext.accounts.doctype.purchase_invoice.purchase_invoice.make_regional_gl_entries': 'erpnext.regional.india.utils.make_regional_gl_entries',
```
not sure why this is showing up in diff.

18alantom replied:
There was a merge conflict when merging `version-13-hotfix` into this. What should be done now?

18alantom committed:
refactor: reduce steps, reword cards

18alantom committed:
Merge branch 'version-13-hotfix' into update-stock-onboarding

18alantom committed:
fix: minor changes

- remove Is Group from warehouse
- change stock entry type
- link to stock entry type
- add posting date to stock reco
- change report to Stock Projected Qty
- highlight quality inspection action
- remove allow neg highlight

18alantom committed:
refactor: use Form Tour doc instead of controller form tour

note - keeping controller form tours as a fallback, new form tours
seem to work only for Stock Settings

18alantom committed:
Merge branch 'version-13-hotfix' into update-stock-onboarding

18alantom commented:
**Todo**
- [ ] Warehouse slide routing: Warehouse Tree → New Warehouse  → Warehouse Form Tour  → Save

18alantom committed:
fix: rename form tours to doctype names, remove tours from js controllers

18alantom committed:
Merge branch 'update-stock-onboarding' of https://github.com/18alantom/erpnext into update-stock-onboarding

18alantom committed:
fix: re-order tour to circumvent glitchy save highlight
